### ISSUE TITLE
ICE on Struct type aliases

### ISSUE THREAD

**Issue by alexvitkov** [noir-lang/noir#2502]
### Aim

This code panics with an internal compiler error:
```rust
struct Foo {}
type StillFoo = Foo;

fn main() {
    let foo: StillFoo = Foo {};
}
```

It's caused by the fact that we resolve type aliases before structs -- this example can be fixed by changing the resolution order (calling `resolve_type_aliases` before `resolve_structs`), however we can have both structs referencing typedefs and typedefs referencing structs, for which there's no trivial fix:

```rust
struct Foo {}
type StillFoo = Foo;

struct Bar {
    foo: StillFoo,
}

fn main() {
    let bar = Bar { foo: Foo{} };
}
```

### Expected Behavior

The examples above work

### Bug

ICE

### To Reproduce

1.
2.
3.
4.

### Installation Method

Compiled from source

### Would you like to submit a PR for this Issue?

Yes

**Comment by yordanmadzhunkov** [comment]
yes, same here https://github.com/metacraft-labs/noir/tree/yordan/type_alias_panic
